src/lib.rs

```diff
@@ -0,0 +1,42 @@
+mod waves;
+pub mod seed_generator;
+
+use std::collections::HashMap;
+use std::thread;
+use std::time::Duration;
+use std::thread::JoinHandle;
+
+pub fn generate_seeds(threads_count: u32, looking_for: &[String]) {
+    let mut worlds: HashMap<String, usize> = HashMap::new();
+    for looking in looking_for.iter() {
+        worlds.insert(looking.to_string(), looking.len());
+    }
+
+    let mut children: Vec<JoinHandle<()>> = vec![];
+    for _i in 0..threads_count {
+        let handler = spawn_thread_to_generate_seeds(&worlds);
+        children.push(handler);
+    };
+    thread::sleep(Duration::from_secs(1));
+    println!("----------");
+
+    for child in children {
+        let _ = child.join();
+    }
+}
+
+fn spawn_thread_to_generate_seeds(worlds: &HashMap<String, usize>) -> JoinHandle<()> {
+    let mut map_copy: HashMap<String, usize> = HashMap::new();
+    HashMap::clone_from(&mut map_copy, &worlds);
```

Comment:
Also I want to ask - is there are better way to share data between threads. I do not to copy HashMap actually, I know that it will be use only for reading, and I can save some heap space and use only one hasmap here. Why i can't do that?

Reply:
You can probably refactor to use a &HashMap (sharable reference).